pkg/config/setup/config.go

```diff
@@ -1086,6 +1086,11 @@ func InitConfig(config pkgconfigmodel.Config) {
 	// The histogram buckets use to track the time in nanoseconds it takes for a DogStatsD listeners to push data to the server
 	config.BindEnvAndSetDefault("telemetry.dogstatsd.listeners_channel_latency_buckets", []string{})
 
+	// Agent Telemetry
+	// Enable various telemetry on the internals of the Agent.
+	// This will not tirgger any billable custom metrics.
+	config.BindEnvAndSetDefault("agent_telemetry.enabled", false)
```

Comment:
I do not have strong preference. When I started spend some time thinking of reusing telemetry name space or package but in the end decided against it due to drastically different semantics and desire to avoid breaking existing code and behavior. It is also a different code hierarchy now. I do hope that at some point I would be able to to merge either agent_telemetry into telemetry package/component or other way around.

Reply:
I am closing for now. We can always revisit it later.